### PR TITLE
Post - Mastodon scheduler

### PR DESCRIPTION
**Pull request by steffilazerte**
Draft post

* [X] I have read the Content Guidelines.
* [X] I have read the Technical Guidelines.
* [X] I used or followed the R Markdown or Markdown template.
* [X] I have followed the Style Guide.
* [X] I created or updated my author metadata with correct folder name.
* [X] I have added relevant tags after browsing existing tags (including "community" tag).
* [X] I have added the "tech notes" tag if this is a technote.
* [X] I ran `roblog::ro_lint_md()` on index.md (optional).
* [X] I ran `roblog::ro_check_urls()` on index.md (optional).
* [X] I ran a spell-check on index.md.

### CODE REVIEW

**Comment by steffilazerte** [comment]
Okay, @yabellini, @maelle, would you two mind taking a quick look/review of my post?

All feedback welcome!

**Comment by yabellini** [comment]
I like it very much.  :-) 
I think I would add that the post queue is important when you work in a team, because it allows you to see the schedule of posts.  Even though I usually use the web to post I pay attention to the issues that are scheduled and it helps me to better decide the timing.

File: content/blog/2023-05-17-scheduling-mastodon/index.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+slug: "scheduling-mastodon"
+title: Scheduling Mastodon Posts with rtoot and GitHub Actions
+author:
+  - Steffi LaZerte
+date: 2023-05-17
+tags:
+  - tech notes
+  - Community
+  - GitHub Actions
+  - Community Manager Tools
```
Comment (maelle) [inline content/blog/2023-05-17-scheduling-mastodon/index.md:11]: add social media, and mastodon?

File: content/blog/2023-05-17-scheduling-mastodon/index.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+description: "How to schedule Mastodon Posts using rtoot and GitHub Actions"
+---
+
+A lot of what I do as rOpenSci's community assistant is related to social media.
```
Comment (maelle) [inline content/blog/2023-05-17-scheduling-mastodon/index.md:15]: now realizing we never had a post introducing you :crying_cat_face:

File: content/blog/2023-05-17-scheduling-mastodon/index.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+and did not
+
+- require us to give Mastodon credentials to an unknown third party
+- require a poster to have access to the main account credentials
+- cost much (or ideally anything)
+
+This led us to create a workflow for scheduling Mastodon posts using R, 
+[rtoot](https://schochastics.github.io/rtoot), and GitHub actions 🎉!
+
+We hope to continue sharing examples of our Community Management Tools so you
```
Comment (maelle) [inline content/blog/2023-05-17-scheduling-mastodon/index.md:49]: I'd put this in the second part of the sentence "You can find an example blablabla where we hope to continue sharing further examples blablabla"

File: content/blog/2023-05-17-scheduling-mastodon/index.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+can find an 
+[example of this workflow](https://github.com/ropensci-org/ro-cmtoolkit/tree/main/scheduled_socials_example) 
+in our new 
+[`ro-cmtools` repository](https://github.com/ropensci-org/ro-cmtoolkit/). 
+
+### What does this workflow entail?
```
Comment (maelle) [inline content/blog/2023-05-17-scheduling-mastodon/index.md:55]:
```suggestion
### What does this workflow entail?

```

File: content/blog/2023-05-17-scheduling-mastodon/index.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+- Using a private repository on GitHub (Mastodon credentials are stored as GitHub Secrets)
```
Comment (maelle) [inline content/blog/2023-05-17-scheduling-mastodon/index.md:56]: share what these credentials are? are they a token, a password?

File: content/blog/2023-05-17-scheduling-mastodon/index.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+- Schedule posts in the future
+- Allow people to post without giving them admin access to the Mastodon account
+- Keep track of old posts (closed issues)
+- Modify scheduled posts
+- Use issue templates to create post templates
+- Post from anywhere with an Internet connection (no R setup required)
+- Specify different time zones to make posting across time zones simpler ([in theory at least](https://fosstodon.org/@ropensci/109458205543474658))
+- Embrace flexibility, we can add/change YAML keys to cue any specific behaviour required[^3]
+
+[^3]: For example, I've been toying with the idea of adding a YAML key for post id,
```
Comment (maelle) [inline content/blog/2023-05-17-scheduling-mastodon/index.md:81]: maybe have a paragraph about future work, with this and the scheduler footnote?

File: content/blog/2023-05-17-scheduling-mastodon/index.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+### Why not use this workflow?
+
+This workflow does have some limitations
+
+- **It can be fragile**   
+  If someone creates a post with the incorrect metadata it will fail, 
+  but only when run, not while writing the post, which can be frustrating. 
+- **Interactions**  
+  While not impossible[^4], it can be tricky or at least a bit clunky to 
+  interact with existing posts using this workflow.
```
Comment (maelle) [inline content/blog/2023-05-17-scheduling-mastodon/index.md:93]: could this be consolidated with the other remark on using post ID to reply?

File: content/blog/2023-05-17-scheduling-mastodon/index.md [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+- **It can burn GitHub Action Minutes**  
+  If you set the CRON to run every hour, this might burn through more GitHub
+  Action minutes than you would like, even with caching. This is why we set it
```
Comment (maelle) [inline content/blog/2023-05-17-scheduling-mastodon/index.md:96]: link to GHA pricing?

File: content/blog/2023-05-17-scheduling-mastodon/index.md [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+[API query](https://docs.joinmastodon.org/methods/statuses/).
+
+
+The Mastodon API and `rtoot::post_toot()` do have options to schedule posts, 
+but I'm unclear how easy it is to see the queue. One thing I like about our
+existing workflow is that I can easily see and modify the queue.
+
+And that's that! There are definitely other options out there, but I have to admit that
+I thoroughly enjoyed the experience of making our own. Even if testing it was a 
+bit [terrifying]() 😉
```
Comment (maelle) [inline content/blog/2023-05-17-scheduling-mastodon/index.md:110]: missing link

File: content/blog/2023-05-17-scheduling-mastodon/index.md [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+
+{{< figure src = "toot_sm.png" width = "500" alt = "A screen shot of a Mastodon post by Steffi LaZerte with a photo of a cat in a canoe looking at trees on a distance short. The text states 'Hi! Hopefully a final test of rtoot (https://github.com/schochastics/rtoot) and this time scheduled via GitHub actions 😱 Oh the thrill of accidentally sending a million toots 😉 And because there are never enough kitties, a photo of Vivi canoe camping! #RStats'" class = "center">}}
+
+
+
```
Comment (maelle) [inline content/blog/2023-05-17-scheduling-mastodon/index.md:115]: add some sort of conclusion? that's where the future work part could go actually.

**Review by maelle — COMMENTED** [review]
awesome work and very cool to see it written up @steffilazerte! As you yourself would say, my comments are mostly suggestions. :wink:

File: content/blog/2023-05-17-scheduling-mastodon/index.md [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+We've enjoyed exploring this new community, but unfortunately there aren't as many tools for working with
+Mastodon posts as there are for Twitter. There are some schedulers available, 
+but we thought this might be an opportunity to develop one of our own; 
+one that worked programatically and allowed us to design it the way we wanted it to work. 
+
+In particular, we wanted a workflow that allowed us to...
+
+- schedule multiple posts at a time
+- easily see and modify these posts
+- have posts with images *and* Alt text
```
[the remark this answers is not in the document]
Comment (maelle) [reply content/blog/2023-05-17-scheduling-mastodon/index.md:37]: should there even be a link to a guide about alternative text? https://axesslab.com/alt-texts/

File: content/blog/2023-05-17-scheduling-mastodon/index.md (hunk 2 above)
Comment (steffilazerte) [reply content/blog/2023-05-17-scheduling-mastodon/index.md:15]: Hahah, that's okay. I think I just really gradually came on 😉

File: content/blog/2023-05-17-scheduling-mastodon/index.md [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+A lot of what I do as rOpenSci's community assistant is related to social media.
+I check for posts about rOpenSci packages, invite people to share [usecases](/usecases),
+advertise [upcoming events](/events), as well as promoting
+[new packages which completed software peer review](/software-review).
+
+Up [until this past year](/blog/2022/11/16/mastodon-en/) most of this work was 
+focused on Twitter and because we want to reach an international audience, 
+we used Tweetdeck to schedule posts for different timezones[^1]. 
+
+[^1]: I already get up early for the European Central Coworking session and am
```
[the remark this answers is not in the document]
Comment (steffilazerte) [reply content/blog/2023-05-17-scheduling-mastodon/index.md:24]: Very good point, thanks!

File: content/blog/2023-05-17-scheduling-mastodon/index.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (steffilazerte) [reply content/blog/2023-05-17-scheduling-mastodon/index.md:37]: That's a great guide, we should add it to the Blog Guide!

File: content/blog/2023-05-17-scheduling-mastodon/index.md (hunk 10 above)
Comment (steffilazerte) [reply content/blog/2023-05-17-scheduling-mastodon/index.md:138]: I created the 'future ideas' section as you suggest (great idea, it worked well I think), but left the comment about enjoying the process as a bit of a concluding sentence. What do you think? Is that still abrupt?

**Comment by steffilazerte** [comment]
Thank you both! I've incorporated your suggestions, and, if you have the time, I'd appreciate a final look at the end as I've consolidated the 'future ideas' in to a section with the concluding sentence at the end. Let me know if that works, or seems too abrupt.

File: content/blog/2023-05-17-scheduling-mastodon/index.md [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+
+Right now our workflow doesn't have the capability to interact with posts on 
+Mastodon. However, I've been toying with the idea of adding a YAML key for post id, 
+so we could set up scheduled replies to existing posts.
+
+Boosting and favouriting seem possible, but would probably require using 
+`rtoot::rtoot()` to create a custom 
+[API queries](https://docs.joinmastodon.org/methods/statuses/). For this workflow
+that may be a bit too clunky, but it could be possible to set up a different type
+of issue template where post ids to be boosted/favourtied could be listed.
```
Comment (maelle) [inline content/blog/2023-05-17-scheduling-mastodon/index.md:122]:
```suggestion
of issue template where post ids to be boosted/favourited could be listed.
```

**Comment by maelle** [comment]
it looks great!

**Comment by steffilazerte** [comment]
Thanks @maelle!

**Pull request merged**